src/core/rtls_core_generate_certificate.c

```diff
@@ -64,11 +65,88 @@ rats_tls_err_t rtls_core_generate_certificate(rtls_core_context_t *ctx)
 			.common_name = (const unsigned char *)"RATS-TLS",
 		},
 	};
-	enclave_attester_err_t q_err;
-	q_err = ctx->attester->opts->collect_evidence(ctx->attester, &cert_info.evidence,
-						      ctx->config.cert_algo, hash, hash_size);
-	if (q_err != ENCLAVE_ATTESTER_ERR_NONE)
-		return c_err;
+	attestation_evidence_t evidence;
+	memset(&evidence, 0, sizeof(attestation_evidence_t));
+
+	// TODO: implement per-session freshness and put "nonce" in custom claims list.
+	uint8_t *claims_buffer = NULL;
+	size_t claims_buffer_size = 0;
+
+	/* Check if we need to create claims_buffer. */
+	/* For sha256, the length of pubkey-hash-value is 36 bytes, so here we assume that the size of user-data field is
+	 * always larger than the length of pubkey-hash-value. */
+	// TODO: We need a way to check the max length of user-data that attester accept.
+	if (ctx->config.custom_claims_length) {
```

Comment:
This logic can be simplified. Please correct me if I'm wrong.
Regardless of whether the user has set custom claims or not, the code should 
1. Take the `pubkeyy hash` as one of the items and insert it into the claims.
2. Then calculate the hash of the claims as the input of collect_evidence.
There is no need to check whether `if (ctx->config.custom_claims_length) `

Reply:
![image](https://user-images.githubusercontent.com/26923638/206444133-c32fce6e-99eb-4dbe-8086-8f1129cb54fd.png)

I re-read the proposal and draw the pic, thanks.